docs/keps/20200619-kubefed-pull-reconciliation.md

```diff
@@ -0,0 +1,362 @@
+cluster.
+
+The resources can be filtered by the label `kubefed.io/managed=true` to exclude them from the rest of Kubernetes resources.
+In addition to that, the kubefed daemon periodically checks the type of the federated
+resources and create an `Informer` per `FederatedTypeConfig`.
+The list of `FederatedTypeConfig` available resources is defined in the control plane cluster.
+There is no need to create an `Informer` per kind of resource type in the Kubernetes cluster.
+The `Informer` populates the cache with the status of the federated resources.
+
+In order to know the `TargetType` of the federated resources, the control-plane allows daemons
+to `get` the list of federated types.
+
+
+#### Desired State Reconciliation
+
+Every daemon is responsible of keeping on sync which federated resources need to be created/updated/deleted
+in its managed cluster.
+
+To do so, the kubefed talks to the control-plane to be informed of which resources
+need to be federated in the cluster.
+
+Likewise this reconciliation loop is in charge of reverting any local changes done in a cluster to
+a federated resource.
+This ensures that the desired state defined in the control-plane cluster is enforced
+in the target clusters.
+
+In the `kubefed control-plane` section, we present the alternative pull-based reconciliation models
+and how the daemon and control-plane work together to enforce the desired state at any time.
+
+### Kubefed control-plane
+
+The control-plane watches the kubefed clusters consuming the exposed endpoints to be aware
+of issues in relation to the clusters and the federated resources.
+
+A critical operation is to constantly watch the status of the federated resources.
+
+<img src="./images/kubefedv2Example.jpg">
+
+When requesting the status of a federated resource, the result needs to report the status
+of the propagation and its own current state.
+
+Another crucial operation is the reconciliation of the desired state, the control-plane
+works as a centralized system where the customer defines what and where a federated resource
+is created.
+Keeping the desired state synced with the state of the clusters would define the success of this new architecture.
+
+A common operation that would be consumed by the daemons consists on getting the list of the available federated types.
+The daemons would be able to `get` the `FederatedTypeConfig` resources, so they
+are aware of the types of resources federated at any time.
+
+Next we present the proposed model to keep the desired state synced across all the cluters.
+
+#### Desired State Reconciliation - Watch Federated types
+
+Every kubefed daemon has to create a remote informer to watch any changes in the federated
+types which all belong to the same group `types.kubefed.io`.
+This remote informer is created against the control-plane cluster, so a kubeconfig
+and the required permissions should be granted to all the kubefed clusters.
```

Comment:
What special permissions are we talking about. Isn't read only access for all federated resources enough?